nx/lib/nx.ex

```diff
@@ -3097,33 +3163,137 @@ defmodule Nx do
         ]
       >
 
+  ## Vectorized tensors
+
+  Vectors remain unchanged, and normal broadcast rules apply otherwise.
+
+      iex> a = Nx.tensor([[[1, 2, 3]], [[4, 5, 6]]]) |> Nx.vectorize(:x)
+      iex> Nx.broadcast(a, {2, 3})
+      #Nx.Tensor<
+        vectorized[x: 2]
+        s64[2][3]
+        [
+          [
+            [1, 2, 3],
+            [1, 2, 3]
+          ],
+          [
+            [4, 5, 6],
+            [4, 5, 6]
+          ]
+        ]
+      >
+
+  For tensor inputs, the broadcast will only take the shape in consideration.
+
+      iex> a = Nx.tensor([[1, 2, 3], [4, 5, 6]]) |> Nx.vectorize(:x)
+      #Nx.Tensor<
+        vectorized[x: 2]
+        s64[3]
+        [
+          [1, 2, 3],
+          [4, 5, 6]
+        ]
+      >
+      iex> b = Nx.tensor([[[1, 2, 3], [4, 5, 6]]], names: [nil, nil, :y]) |> Nx.vectorize(:a)
+      #Nx.Tensor<
+        vectorized[a: 1]
+        s64[2][y: 3]
+        [
+          [
+            [1, 2, 3],
+            [4, 5, 6]
+          ]
+        ]
+      >
+      iex> Nx.broadcast(a, b, axes: [1], names: [:i, :j])
```

Comment:
Are we sure we don't want to vectorize the result with the `:a` as well?

Reply:
I'm never sure haha!
Though we will have broadcast_vectors, so this behavior allows for a bit more flexibility

Reply:
good point, so let's ship this!